c10/core/CPUAllocator.cpp

```diff
@@ -153,6 +153,8 @@ void SetCPUAllocator(at::Allocator* alloc) {
   SetAllocator(DeviceType::CPU, alloc);
 }
 
+#ifndef C10_MOBILE
+
```

Comment:
I think it will be better to have the ifdef inside of this function, instead of having a separate mobile specific def of this function. This function being `GetDefaultCPUAllocator`, no?

Reply:
Makese sense.  I'm also a bit concerned about the `REGISTER_ALLOCATOR(DeviceType::CPU, &g_mobile_cpu_allocator);` in the other file because registering the allocator for a given device in different files is difficult to keep track of.  The change you're suggesting would eliminate that issue as well, right?

Reply:
g_mobile_cpu_allocator is not visible in this file.  The class that this is an instance of is not available here either.  I can put all implementation details into this file if that's what you guys prefer.

Reply:
(by which I mean removing MobileCPUAllocator and copying its contents here.)

Reply:
I think moving the implementation is fine since it is small and has no other dependencies.  But you could also call GetDefaultMobileCPUAllocator from this file instead, right?

Reply:
> g_mobile_cpu_allocator is not visible in this file. The class that this is an instance of is not available here either. I can put all implementation details into this file if that's what you guys prefer.

Is it possible to do that by including the definition in MobileAllocator.h and including that here? And also doing this `DefaultMobileCPUAllocator<std::max(8u, Configuration::kAlignment), 16u> g_mobile_cpu_allocator;` in this file inside `#ifdef C10_MOBILE`. That way static instance is created only for mobile build. Then we should be able to move ifdef inside `GetDefaultCPUAllocator` function. What do you think?

Reply:
g_mobile_cpu_allocator is not really mobile specific since QNNPACK and XNNPACK are not.  In other words, QNNPACK and XNNPACK can be used on non-mobile builds making the presence of this allocator a requirement.

Reply:
> I think moving the implementation is fine since it is small and has no other dependencies. But you could also call GetDefaultMobileCPUAllocator from this file instead, right?

Right you could call the function but not register the allocator since it's got internal linkage. Won't be an issue if I move the code. Let me know which you guys prefer.  No strong preference here.

Reply:
> g_mobile_cpu_allocator is not really mobile specific since QNNPACK and XNNPACK are not. In other words, QNNPACK and XNNPACK can be used on non-mobile builds making the presence of this allocator a requirement.

I see.So can we still do `DefaultMobileCPUAllocator<std::max(8u, Configuration::kAlignment), 16u> g_mobile_cpu_allocator; ` outside of ifdef? Upto you. Just sharing what came to my mind.

Reply:
I think moving the code is fine.  I think putting `g_mobile_cpu_allocator` outside of ifdef is also fine.  The only thing I don't think is fine is having `REGISTER_ALLOCATOR(DeviceType::CPU,` show up in two different files.

Reply:
Sounds good.  Will move everything into one file.

Reply:
Done. Marking as resolved.